pkg/server/util/regentryutil/cache.go

```diff
@@ -0,0 +1,61 @@
+package regentryutil
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/spiffe/spire/proto/spire/common"
+)
+
+// FetchRegistrationEntriesCache is a wrapper around LRU cache with expiry, used for caching registration entries of a agent
+type FetchRegistrationEntriesCache struct {
+	Cache   *lru.Cache
+	TimeNow func() time.Time
+
+	mu sync.RWMutex
+}
+
+type cacheValue struct {
+	entries   []*common.RegistrationEntry
+	expiresAt time.Time
+}
+
+func NewFetchX509SVIDCache(cacheSize int) (*FetchRegistrationEntriesCache, error) {
+	cache, err := lru.New(cacheSize)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create lru cache: %v", err)
```

Comment:
Please use field-based logging here to avoid high-cardinality log messages

Reply:
Also need to capitalize log message

Reply:
There actually no log here